### PR TITLE
Add simple rate limiting util

### PR DESCRIPTION
**Pull request by gcapalbo**
@calellowitz

### CODE REVIEW

**Comment by czue** [comment]
looks good! might be nice to have a few tests

**Comment by gcapalbo** [comment]
@czue good point, updated

File: dimagi/utils/tests/test_rate_limit.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        iteration_count = 0
+
+        while (datetime.utcnow() - start) < timedelta(seconds=18):
+            # Only allow 10 actions every 5 seconds in an 18 second period of time
+            if rate_limit('rate-limit-test', actions_allowed=10, how_often=5):
+                rate_limit_count += 1
+            iteration_count += 1
+
+        self.assertEqual(rate_limit_count, 40)
+        self.assertGreater(iteration_count, 40)
```
Comment (czue) [inline dimagi/utils/tests/test_rate_limit.py]: nice

**Comment by gcapalbo** [comment]
@dimagi/team-commcare-hq fyi, in case you ever have to do rate limiting, this is a util which can help

File: dimagi/utils/tests/test_rate_limit.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    def test_rate_limit(self):
+        start = datetime.utcnow()
+        rate_limit_count = 0
+        iteration_count = 0
+
+        while (datetime.utcnow() - start) < timedelta(seconds=18):
+            # Only allow 10 actions every 5 seconds in an 18 second period of time
+            if rate_limit('rate-limit-test', actions_allowed=10, how_often=5):
+                rate_limit_count += 1
+            iteration_count += 1
```
Comment (millerdev) [inline dimagi/utils/tests/test_rate_limit.py]: Does this loop run for 18 seconds when running tests?

Comment (gcapalbo) [inline dimagi/utils/tests/test_rate_limit.py]: yeah

Comment (millerdev) [inline dimagi/utils/tests/test_rate_limit.py]: Might want to rewrite this to patch the time functions so you can make the test complete way faster. Ping me if you want ideas about how to do that.

Comment (gcapalbo) [inline dimagi/utils/tests/test_rate_limit.py]: I'm not sure if that would work - the expiration is set in redis and I don't think we'd be able to patch that.  If we wanted to reduce the test time, I could also just choose smaller numbers here.

Comment (millerdev) [inline dimagi/utils/tests/test_rate_limit.py]: ah, forgot about the redis timeout. smaller numbers should be good enough

**Pull request merged**